### PR TITLE
For compatibility, avoid the new fast path if the incoming pointer is 0

### PR DESCRIPTION
**Pull request by Catfish-Man**
Fixes rdar://problem/60882075

### CODE REVIEW

**Comment by Catfish-Man** [comment]
@swift-ci please test

File: stdlib/public/Darwin/Foundation/NSStringAPI.swift [hunk 1]
```diff
@@ -178,7 +178,10 @@ extension String {
   /// Creates a string by copying the data from a given
   /// C array of UTF8-encoded bytes.
   public init?(utf8String bytes: UnsafePointer<CChar>) {
-    if let str = String(validatingUTF8: bytes) {
+    // This isn't declared to accept nil but in practice it appears some people
+    // manage to pass it nil anyway
+    if let str = String(validatingUTF8: bytes),
+       unsafeBitCast(bytes, to: UInt.self) != 0 {
```
Comment (milseman) [inline stdlib/public/Darwin/Foundation/NSStringAPI.swift:184]: We really need to curb some of our excessive unsafe bitcasts here. `UInt(bitPattern: bytes)` should work.

**Review by milseman — APPROVED** [review]
Get rid of the unsafe cast and LGTM

**Comment by Catfish-Man** [comment]
@swift-ci please test and merge

**Pull request closed without merging**